service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -137,14 +143,96 @@ public PageableAdvancedDto<EventDto> getAll(Pageable page, Principal principal)
-        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
-        setSubscribes(events, eventDtos, attender);
+        List<Event> events = sortUserEventsByEventType(eventType, attender, userLatitude, userLongitude);
+        Page<Event> eventPage = new PageImpl<>(events, page, events.size());
+        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(eventPage);
+        setSubscribes(eventPage, eventDtos, attender);
         return eventDtos;
     }
 
+    private List<Event> sortUserEventsByEventType(
+        String eventType, User attender, String userLatitude, String userLongitude) {
+        if (eventType.equalsIgnoreCase("ONLINE")) {
+            return getOnlineUserEventsSortedByDate(attender);
+        }
+
+        if (eventType.equalsIgnoreCase("OFFLINE")) {
+            return (!userLatitude.isBlank() && !userLongitude.isBlank())
+                ? getOfflineUserEventsSortedCloserToUserLocation(attender, userLatitude, userLongitude)
+                : getOfflineUserEventsSortedByDate(attender);
+        }
+        return eventRepo.findAllByAttender(attender.getId()).stream().sorted(getComparatorByDates())
+            .collect(Collectors.toList());
+    }
+
+    private List<Event> getOnlineUserEventsSortedByDate(User attender) {
+        return eventRepo.findAllByAttender(attender.getId()).stream()
+            .filter(event -> event.getEventType().equals(EventType.ONLINE)
+                || event.getEventType().equals(EventType.ONLINE_OFFLINE))
+            .sorted(getComparatorByDates())
+            .collect(Collectors.toList());
+    }
+
+    private List<Event> getOfflineUserEventsSortedByDate(User attender) {
+        return eventRepo.findAllByAttender(attender.getId()).stream()
+            .filter(event -> event.getEventType().equals(EventType.OFFLINE)
+                || event.getEventType().equals(EventType.ONLINE_OFFLINE))
+            .sorted(getComparatorByDates())
+            .collect(Collectors.toList());
+    }
+
+    private List<Event> getOfflineUserEventsSortedCloserToUserLocation(
+        User attender, String userLatitude, String userLongitude) {
+        List<Event> eventsFurtherSorted = eventRepo.findAllByAttender(attender.getId()).stream()
+            .filter(event -> event.getEventType().equals(EventType.OFFLINE)
+                || event.getEventType().equals(EventType.ONLINE_OFFLINE))
+            .sorted(getComparatorByDistance(Double.parseDouble(userLatitude), Double.parseDouble(userLongitude)))
+            .filter(this::isEventRelevant)
+            .collect(Collectors.toList());
+        List<Event> eventsPassed = getOfflineUserEventsSortedByDate(attender).stream()
+            .filter(event -> !isEventRelevant(event))
+            .collect(Collectors.toList());
+        eventsFurtherSorted.addAll(eventsPassed);
+        return eventsFurtherSorted;
+    }
+
+    private boolean isEventRelevant(Event event) {
```

Comment:
move this to the Event class

Reply:
I have moved to Event class method and named it isRelevant()